### PR TITLE
Add Sapling support to z_validateaddress

### PR DESCRIPTION
**Pull request by str4d**
Part of #3217.

### CODE REVIEW

File: src/rpcmisc.cpp [hunk 1]
```diff
@@ -206,6 +206,37 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
...
+        return obj;
+    }
+
+    UniValue operator()(const libzcash::SaplingPaymentAddress &zaddr) const {
+        UniValue obj(UniValue::VOBJ);
+        obj.push_back(Pair("type", "sapling"));
+#ifdef ENABLE_WALLET
+        if (pwalletMain) {
+            // TODO uncomment when #3273 is merged
+            // obj.push_back(Pair("ismine", pwalletMain->HaveSaplingSpendingKey(zaddr)));
```
Comment (arcalinea) [inline src/rpcmisc.cpp]: `HaveSaplingSpendingKey` will need to take the full viewing key and not a zaddr as an arg

**Comment by str4d** [comment]
Rebased on master and finished, now that #3273 is merged.

**Comment by str4d** [comment]
@zkbot try

File: src/rpcmisc.cpp [hunk 2]
```diff
@@ -206,6 +206,43 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
...
+        obj.push_back(Pair("transmissionkey", zaddr.pk_enc.GetHex()));
+#ifdef ENABLE_WALLET
+        if (pwalletMain) {
+            obj.push_back(Pair("ismine", pwalletMain->HaveSpendingKey(zaddr)));
+        }
+#endif
+        return obj;
+    }
+
+    UniValue operator()(const libzcash::SaplingPaymentAddress &zaddr) const {
```
Comment (arcalinea) [inline src/rpcmisc.cpp]: `doc/payment-api.md` will have to be modified to document what gets returned for Sapling addresses, and to add the `type` field to output.

Non-blocking, can do this in follow-up PR

**Comment by bitcartel** [comment]
@zkbot r+

**Pull request merged**